http-client/src/main/java/io/micronaut/http/client/BasicAuth.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.client;
+
+import io.micronaut.context.annotation.Prototype;
+
+/**
+ * A class representing basic auth authorization header.
+ * To be used in {@link io.micronaut.http.client.annotation.Client}
+ *
+ * @author Ashwini Mutalik Desai
+ */
+@Prototype
+public final class BasicAuth {
```

Comment:
I don't think this should be annotated with `@Prototype` and if it is going to be need each parameter will be needed to be annotated with `@Parameter` although since it is not used as a bean I don't think having `@Prototype` is needed at all.

This raises another question about whether we should support this type on the server as well and have a `RequestArgumentBinder` that allows you to bind the `BasicAuth` type to method arguments in the controller. If it is a agreed that that is the case then this should not be in the `client` package.

Reply:
Yes, you are right about ```@Prototype```. Removed it.

About making this available as method arguments in the controller, I don't think it will of much help since request will have basic auth already encoded and available as a string. It could be may be used in something like ```AuthenticationProvider```. If that is the use case we are thinking about, we can do it.